crates/uv/src/commands/project/lock.rs

```diff
@@ -246,97 +246,139 @@ pub(super) enum LockMode<'env> {
     Frozen,
 }
 
-/// Perform a lock operation, respecting the `--locked` and `--frozen` parameters.
-#[allow(clippy::fn_params_excessive_bools)]
-pub(super) async fn do_safe_lock(
-    mode: LockMode<'_>,
-    target: LockTarget<'_>,
-    settings: &ResolverSettings,
-    network_settings: &NetworkSettings,
-    state: &UniversalState,
+/// A lock operation.
+pub(super) struct LockOperation<'env> {
```

Comment:
I didn't want to add a `constraints` argument to `do_safe_lock` when only one caller needed it, so I migrated this to a struct with a builder pattern.